### PR TITLE
Added doc files to sync-in and sync-out

### PR DESCRIPTION
**Pull request by mgc1194**
**What:** Adding Documentation Navigation Menus to the i18n pipeline. A new method `localize_docs` has been added to the sync-in that collects all Navigation Categories from the database and generates a JSON file that will be uploaded to Crowdin. In the sync-out, the localized Navigation Categories are added to dashboard locale.

**Why:** The Documentation pages is not fully translatable and will be necessary as CSD includes multiple Labs in the curriculum.

## Links

- jira ticket: [FND-2062](https://codedotorg.atlassian.net/browse/FND-2062)
- Code.org [docs](https://studio.code.org/docs)

## Testing story

Strings are successfully added to Crowdin:
Captura de Pantalla 2022-08-04 a la(s) 5 40 35 p m

Translated strings are successfully downloaded from Crowdin:
Captura de Pantalla 2022-08-04 a la(s) 5 51 14 p m

Translations successfully appear in Navigation Menu:
Captura de Pantalla 2022-08-08 a la(s) 1 10 21 p m

## Follow-up work

Related tickets:
- jira ticket: [FND-2060](https://codedotorg.atlassian.net/browse/FND-2060)
- jira ticket: [FND-2064](https://codedotorg.atlassian.net/browse/FND-2064)
## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/sync-in.rb [hunk 1]
```diff
@@ -36,6 +37,30 @@ def sync_in
   raise e
 end
 
+# This function localizes all content in studio.code.org/docs
+def localize_docs
+  puts "Preparing docs content"
+  docs_content_file = File.join(I18N_SOURCE_DIR, "docs", "programming_environments.json")
+  programming_env_docs = {}
+  ProgrammingEnvironment.all.each do |env|
+    name = env.name
```
Comment (annaxuphoto) [inline bin/i18n/sync-in.rb:46]: just for my understanding, where does `name` come from?

File: bin/i18n/sync-in.rb [hunk 2]
```diff
@@ -36,6 +37,30 @@ def sync_in
...
+    programming_env_docs[name] = {
+      'name' => env.properties["title"],
+      'description' => env.properties["description"],
+      'categories' => {},
+    }
+    env.categories_for_navigation.each do |category|
```
Comment (annaxuphoto) [inline bin/i18n/sync-in.rb:52]: some documentation on how this structure will look at the end would be super helpful!

**Review by annaxuphoto — APPROVED** [review]
just a couple comments for understanding. This is awesome and super impressive as your first dive into translatability!

File: bin/i18n/sync-in.rb (hunk 1 above)
Comment (mgc1194) [reply bin/i18n/sync-in.rb:46]: Name comes from this file: https://github.com/code-dot-org/code-dot-org/blob/602edd3e55ab732b0df29aaf7adf4f568d854c65/dashboard/config/programming_environments/gamelab.json
The [model](https://github.com/code-dot-org/code-dot-org/blob/602edd3e55ab732b0df29aaf7adf4f568d854c65/dashboard/app/models/programming_environment.rb) reads the file and seeds the database with the json content: 
`  def self.seed_all(glob="config/programming_environments/*.json")
    removed_records = all.pluck(:name)
    Dir.glob(Rails.root.join(glob)).each do |path|
      removed_records -= [ProgrammingEnvironment.seed_record(path)]
    end
    where(name: removed_records).destroy_all
  end `

**Pull request merged**